options/options_type.h

```diff
@@ -123,18 +206,93 @@ class OptionTypeInfo {
+  // Aliases are valid options that are parsed but are not converted to strings
+  // or compared.
+  bool IsAlias() const { return IsEnabled(OptionVerificationType::kAlias); }
+
   bool IsEnabled(OptionVerificationType ovf) const {
     return verification == ovf;
   }
 
+  // Returns the sanity level for comparing the option.
+  // If the options should not be compared, returns None
+  // If the option has a compare flag, returns it.
+  // Otherwise, returns "exact"
+  ConfigOptions::SanityLevel GetSanityLevel() const {
+    if (IsDeprecated() || IsAlias()) {
+      return ConfigOptions::SanityLevel::kSanityLevelNone;
+    } else {
+      auto match = (flags & OptionTypeFlags::kCompareExact);
+      if (match == OptionTypeFlags::kCompareDefault) {
+        return ConfigOptions::SanityLevel::kSanityLevelExactMatch;
+      } else {
+        return (ConfigOptions::SanityLevel)match;
+      }
+    }
+  }
+
+  // Returns true if the option should be serialized.
+  // Options should be serialized if the are not deprecated, aliases,
+  // or marked as "Don't Serialize".
+  bool ShouldSerialize() const {
+    if (IsDeprecated() || IsAlias()) {
+      return false;
+    } else if (IsEnabled(OptionTypeFlags::kDontSerialize)) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
   bool IsByName() const {
     return (verification == OptionVerificationType::kByName ||
             verification == OptionVerificationType::kByNameAllowNull ||
             verification == OptionVerificationType::kByNameAllowFromNull);
   }
 
- protected:
+  // Parses the option in "opt_value" according to the rules of this class
+  // and updates the value at "opt_addr".
+  // On success, Status::OK() is returned.  On failure:
+  // NotFound means the opt_name is not valid for this option
+  // NotSupported means we do not know how to parse the value for this option
+  // InvalidArgument means the opt_value is not valid for this option.
+  Status ParseOption(const ConfigOptions& config_options,
+                     const std::string& opt_name, const std::string& opt_value,
+                     char* opt_addr) const;
+
+  // Serializes the option in "opt_addr" according to the rules of this class
+  // into the value at "opt_value".
+  Status SerializeOption(const ConfigOptions& config_options,
+                         const std::string& opt_name, const char* opt_addr,
+                         std::string* opt_value) const;
+
+  // Compares the "addr1" and "addr2" values according to the rules of this
+  // class and returns true if they match.  On a failed match, mismatch is the
+  // name of the option that failed to match.
+  bool MatchesOption(const ConfigOptions& config_options,
+                     const std::string& opt_name, const char* addr1,
+                     const char* addr2, std::string* mismatch) const;
+
+  // Used to override the match rules for "ByName" options.
+  bool CheckByName(const ConfigOptions& config_options,
```

Comment:
CheckByName is unclear to me. Maybe replace it with another name which shows its usage.

Reply:
Renamed to MatchesByName